lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -7,162 +7,169 @@
+    private static final String ENDOFPROMPT = "<|endofprompt|>";
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 50256);
+        SPECIAL_TOKENS_X50K_BASE = Collections.unmodifiableMap(map);
+    }
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 50256);
+        map.put(FIM_PREFIX, 50281);
+        map.put(FIM_MIDDLE, 50282);
+        map.put(FIM_SUFFIX, 50283);
+        SPECIAL_TOKENS_P50K_EDIT = Collections.unmodifiableMap(map);
+    }
+
+    static {
+        Map<String, Integer> map = new HashMap<>();
+        map.put(ENDOFTEXT, 100257);
+        map.put(FIM_PREFIX, 100258);
+        map.put(FIM_MIDDLE, 100259);
+        map.put(FIM_SUFFIX, 100260);
+        map.put(ENDOFPROMPT, 100276);
+        SPECIAL_TOKENS_CL100K_BASE = Collections.unmodifiableMap(map);
+    }
+
+    private EncodingFactory() {
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the r50k_base encoding.
+     *
+     * @return an {@link Encoding} instance for the r50k_base encoding
+     */
+    public static Encoding r50kBase() {
+        return fromPredefinedParameters(
+                "r50k_base",
+                "'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+",
+                "/com/knuddels/jtokkit/r50k_base.tiktoken",
+                SPECIAL_TOKENS_X50K_BASE,
+                false
+        );
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the p50k_base encoding.
+     *
+     * @return an {@link Encoding} instance for the p50k_base encoding
+     */
+    public static Encoding p50kBase() {
+        return fromPredefinedParameters(
+                "p50k_base",
+                "'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+",
+                "/com/knuddels/jtokkit/p50k_base.tiktoken",
+                SPECIAL_TOKENS_X50K_BASE,
+                false
+        );
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the p50k_edit encoding.
+     *
+     * @return an {@link Encoding} instance for the p50k_edit encoding
+     */
+    public static Encoding p50kEdit() {
+        return fromPredefinedParameters(
+                "p50k_edit",
+                "'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+",
+                "/com/knuddels/jtokkit/p50k_base.tiktoken",
+                SPECIAL_TOKENS_P50K_EDIT,
+                false
+        );
+    }
+
+    /**
+     * Returns an {@link Encoding} instance for the cl100k_base encoding.
+     *
+     * @return an {@link Encoding} instance for the cl100k_base encoding
+     */
+    public static Encoding cl100kBase() {
+        return fromPredefinedParameters(
+                "cl100k_base",
+                "'(?:[sdmt]|ll|ve|re)|[^\r\n\\p{L}\\p{N}]?+\\p{L}+|\\p{N}{1,3}| ?[^\\s\\p{L}\\p{N}]++[\r\n]*|\\s*[\r\n]|\\s+(?!\\S)|\\s+",
```

Comment:
same, but here I've also made a few groups possessive (based on whether they made backtracking less likely, improving the speed measurably):
<img width="804" alt="image" src="https://github.com/knuddelsgmbh/jtokkit/assets/1841944/d001fa29-7310-4c59-ac74-2b1e1f225de1">